### PR TITLE
Fixes a number of p1 issues

### LINKED ISSUE #52: Problem with getElementById on shadowRoot
In a custom element, I'm trying something like this : 

```
this.elementShadow = this.attachShadow({ mode: 'open' });
this.elementShadow.innerHTML = `<div id="my-id"></div>`;
this.elementShadow.getElementById('my-id');
```

It's OK in a browser with native support, but using the polyfill in a browser not supporting ShadowDOM v1 (Firefox for example), it returns `null`.

It works if I do : 
```
this.elementShadow.querySelector('#my-id');
```

### PR DESCRIPTION
**Pull request by sorvell**
Fixes #52, #30, #119, #95, #96, #116.

Most notably the fix for #116 led to refactoring node insertion/removal. Now all DOM is queried for insertion points. If one is found, the shadowRoot insertion point list is immediately updated and it is no longer done async at render time. This allows the insertion/removal code to rely on the shadowRoot insertion list state to decide if an added element should provoke distribution or may be fast path rendered.

### CODE REVIEW

File: tests/smoke/click.html [hunk 1]
```diff
@@ -1,17 +1,17 @@
 <!doctype html>
 <html>
   <head>
-    <script src="../../../webcomponents-platform/webcomponents-platform.js"></script>
-    <script src="../../../template/template.js"></script>
+    <script src="../../bower_components/webcomponents-platform/webcomponents-platform.js"></script>
```
Comment (azakus) [inline tests/smoke/click.html]: Why use `bower_components` here?

File: src/attach-shadow.js [hunk 2]
```diff
@@ -99,6 +102,11 @@ ShadyRoot.prototype._rendererForHost = function() {
   }
 }
 
+ShadyRoot.prototype.forceRender = function() {
```
Comment (azakus) [inline src/attach-shadow.js]: Why is this a public method?

**Comment by azakus** [comment]
LGTM with nits

File: src/attach-shadow.js [hunk 3]
```diff
@@ -280,6 +248,10 @@ ShadyRoot.prototype.removeEventListener = function(type, fn, optionsOrCapture) {
   this.host.removeEventListener(type, fn, optionsOrCapture);
 }
 
+ShadyRoot.prototype.getElementById = function(id) {
```
Comment (valdrinkoshi) [inline src/attach-shadow.js]: This method is already defined in L227

**Comment by sorvell** [comment]
All of these but #95 and #116 were addressed by https://github.com/webcomponents/shadydom/pull/131. These remaining issues will be in a different PR so closing this one.

**Pull request closed without merging**